### PR TITLE
Ops 2851 poc version monitoring

### PR DESCRIPTION
**Pull request by ldahlke**

### CODE REVIEW

File: .github/workflows/dbcmtag.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      - run: echo "$GITHUB_CONTEXT"
+      - uses: olegtarasov/get-tag@v2.1
+        id: tagName
+        with:
+          tagRegex: "(.*)-(.*)"  # Optional. Returns specified group text as tag name. Full tag string is returned if regex is not defined.
+          tagRegexGroup: 1 # Optional. Default is 1.
+      - name: Some other step # Output usage example
+        with:
+          tagname: ${{ steps.tagName.outputs.tag }}
+      # - uses: actions/checkout@v2
```
Comment (dimapin) [inline .github/workflows/dbcmtag.yml:23]: Do we need these commented out code?

File: dbcmetrics/dbcm_logic/dbcversion.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    '''
+    '''
+    def __init__(self, configuration: DBCMConfiguration):
+        self.configuration = configuration
+        # Loop over instances create a Info metrics object for each and store them in a list
+        self.pmc_infos: Dict [Info] = {}
+        instance: DBCMInstance
+        for instance in self.configuration.instances:
+            self.pmc_infos[instance.name] = Info(instance.name, 'Version Information')
+        self.g = Gauge('BrandenburgGauge', 'Version Information', ['app_instance', 'version'])
```
Comment (dimapin) [inline dbcmetrics/dbcm_logic/dbcversion.py:24]: Why it is called BrandenburgGauge here?

File: dbcmetrics/dbcm_logic/dbcversion.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        self.thread = threading.Thread(target=self.run)
+        self.thread.daemon = True
+        self.thread.start()
+
+    def run(self):
+        '''
+        Function that opens the tunnel and wait in a loop until the stop event is send from the main thread
+        '''
+        def do_something():
+            #strversion: str = self.getVersions('Brandenburg')
```
Comment (dimapin) [inline dbcmetrics/dbcm_logic/dbcversion.py:34]: Is this code needed?

File: tagevent.json [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (dimapin) [inline tagevent.json:1]: What is the file for`?

File: dbcmetrics/read_yaml.py [hunk 5]
```diff
@@ -1,7 +1,12 @@
-import yaml
-
-with open('config.yaml', 'r') as file:
+import yaml,os
```
Comment (dimapin) [inline dbcmetrics/read_yaml.py:1]: Is this a test?

File: dbcmetrics/dbcm_logic/dbcversion.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                labels: Dict = self.getInstanceVersions(i)
+                instance_info: Info = self.pmc_infos[i]
+                labels['app_instance'] = i
+                instance_info.info(labels)
+            sleep(self.configuration.version.intervall)
+        while True:
+            do_something()
+    
+    def getVersions(self, instance):
+        url = requests.get("https://brandenburg.cloud/version")
```
Comment (dimapin) [inline dbcmetrics/dbcm_logic/dbcversion.py:49]: What are the hardcoded values for?

**Review by RalfNellesCap — APPROVED** [review]

**Review by dimapin — APPROVED** [review]

File: tagevent.json (hunk 4 above)
Comment (aimee-889) [reply tagevent.json:1]: this should have been used for local testing of the github action, i suppose its not needed in repo

Comment (ldahlke) [reply tagevent.json]: Is now deleted

**Review by manuela-raschke-cap — APPROVED** [review]

**Review by aimee-889 — APPROVED** [review]

File: dbcmetrics/dbcm_config.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+features:
+  # Key/value pais of features dbcmetrics supports
+  version_metrics: [enabled|disabled]
+instances:
+  # list of instances to be monitored
+  - name: "<name used as prefix for the metrics name, e.g. hugo -> hugo_info>"
+    url: "<Full quallified URL which should be monitored>"
+    shortname: "<abbreviated name of the naem>"
+version_metrics:
+  intervall: <value in seconds, how often the version is polled>
```
Comment (simoncolincap) [inline dbcmetrics/dbcm_config.yaml:10]: Interval only has one L in English

Comment (ldahlke) [reply dbcmetrics/dbcm_config.yaml]: ´Corrected

**Pull request merged**